src/transformers/generation/utils.py

```diff
@@ -1549,6 +1547,23 @@ def generate(
                 )
             if generation_config.cache_implementation == "static":
                 model_kwargs["past_key_values"] = self._get_static_cache(batch_size, generation_config.max_length)
+            elif generation_config.cache_implementation == "quantized":
+                if model_kwargs.get("past_key_values") is not None:
+                    if not isinstance(model_kwargs["past_key_values"], QuantCache):
+                        raise ValueError(
+                            f"Setting `QuantCache` and passing {type(model_kwargs['past_key_values'])} for `past_key_values` is not implemented. "
+                            "Either pass `QuantCache` type or do not pass `past_key_values`"
+                        )
+                else:
+                    # get the config with defualt values if user did not pass anything
+                    cache_config = (
+                        generation_config.cache_config if generation_config.cache_config is not None else CacheConfig()
+                    )
+                    model_kwargs["past_key_values"] = QuantCache(
```

Comment:
Here we could raise an error if quanto is not installed and educate users on how to install quanto, what do you think?